### DISCUSSION TITLE
AR scale in quick look not fixed, how to turn rotation off?

### DISCUSSION THREAD

**Discussion by richardrevesz** [google/model-viewer#2402] [Q&A] [answered]
Hi!

I have AR scale set to fixed, and it is working fine on Android, however, on iOS Quick Look I can still scale the model, it even shows the percentage. Is there a way to lock the scale in Quick Look?

Also, I am using wall placement, and realized the rotation is working differently in WebXR and Quick Look:

- In WebXR it rotates around the horizontal axis. While placing on the wall it can be really confusing, it shouldn't be rotating into the wall surface.
- In Quick Look it is better, it rotates around the normal axis of the wall.

However, I would prefer to lock rotation as well. Is there a way to do it?

Best,
Richard

**Comment by milesgreen** [comment]
Regrading the scaling, this might be related to #2405 where fixed scaling isn't working when auto generating the USDZ? If you set an `ios-src` USDZ, does fixed scaling work for you? It does for me.

**Reply by richardrevesz** [reply]
Thank you! Yes, I am using generated USDZ, so it is a known issue.

**Comment by ghost** [comment]
As for disabling rotation there is no supported way to disable *only*
rotation.

**Reply by richardrevesz** [reply]
What do you mean *only* rotation? I need rotation and scale to be locked, is that possible?

**Comment by elalish** [comment, marked as answer]
> * In WebXR it rotates around the horizontal axis. While placing on the wall it can be really confusing, it shouldn't be rotating into the wall surface.

Oops, that's a bug I can repro. We weren't intending to support rotation for wall placement at all; looks like it's disabled for one-finger rotation but not for two. We'll get that fixed.

**Reply by elalish** [reply]
The fix: https://github.com/google/model-viewer/pull/2412
